include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -43,62 +43,76 @@ template <bool _IsGPU, typename... _Name>
+    {
+        auto __transform_pattern = unseq_backend::transform_reduce_seq<_ExecutionPolicy, _ReduceOp, _TransformOp, _Tp>{
+            __reduce_op, _TransformOp{__transform_op}};
+        auto __reduce_pattern = unseq_backend::reduce_over_group<_ExecutionPolicy, _ReduceOp, _Tp>{__reduce_op};
+
+        sycl::buffer<_Tp> __res(sycl::range<1>(1));
+
+        sycl::event __reduce_event = __exec.queue().submit([&, __n](sycl::handler& __cgh) {
+            oneapi::dpl::__ranges::__require_access(__cgh, __rngs...); // get an access to data under SYCL buffer
+            auto __res_acc = __res.template get_access<access_mode::write>(__cgh);
+            __cgh.single_task<_Name...>([=] {
+                _Tp __result = __transform_pattern(__n, __rngs...);
+                __reduce_pattern.apply_init(__init, __result);
+                __res_acc[0] = __result;
+            });
+        });
+
+        return __future(__reduce_event, __res);
+    }
+};
+
 template <typename _Tp, typename _ReduceOp, typename _TransformOp, typename _ExecutionPolicy, typename _Size,
           typename _InitType, oneapi::dpl::__internal::__enable_if_device_execution_policy<_ExecutionPolicy, int> = 0,
           typename... _Ranges>
 auto
-__parallel_transform_reduce_seq_submitter(_ExecutionPolicy&& __exec, _Size __n, _ReduceOp __reduce_op,
-                                          _TransformOp __transform_op, _InitType __init, _Ranges&&... __rngs)
+__parallel_transform_reduce_seq_impl(_ExecutionPolicy&& __exec, _Size __n, _ReduceOp __reduce_op,
+                                     _TransformOp __transform_op, _InitType __init, _Ranges&&... __rngs)
 {
     using _Policy = typename ::std::decay<_ExecutionPolicy>::type;
     using _CustomName = typename _Policy::kernel_name;
     using _ReduceKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__reduce_seq_kernel, _CustomName,
-                                                                               _ReduceOp, _TransformOp, _Ranges...>;
-
-    auto __transform_pattern = unseq_backend::transform_reduce_seq<_ExecutionPolicy, _ReduceOp, _TransformOp, _Tp>{
-        __reduce_op, _TransformOp{__transform_op}};
-    auto __reduce_pattern = unseq_backend::reduce_over_group<_ExecutionPolicy, _ReduceOp, _Tp>{__reduce_op};
-
-    sycl::buffer<_Tp> __res(sycl::range<1>(1));
-
-    sycl::event __reduce_event = __exec.queue().submit([&, __n](sycl::handler& __cgh) {
-        oneapi::dpl::__ranges::__require_access(__cgh, __rngs...); // get an access to data under SYCL buffer
-        auto __res_acc = __res.template get_access<access_mode::write>(__cgh);
-        __cgh.single_task<_ReduceKernel>([=] {
-            _Tp __result = __transform_pattern(__n, __rngs...);
-            __reduce_pattern.apply_init(__init, __result);
-            __res_acc[0] = __result;
-        });
-    });
+        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__reduce_seq_kernel<_CustomName>>;
```

Comment:
It seems a custom (user) name checking is broken here...
See https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h#L166

Reply:
It doesn't due to the specialization below